### ISSUE TITLE
bug: Exception in ASGI application

### ISSUE THREAD

**Issue by maxfreu** [bentoml/OpenLLM#34]
### Describe the bug

Just gave your repo a quick shot, on 0.1.6 I did
```
openllm download falcon
openllm start falcon
openllm query ...
```
Hardware: 2x1080, 96GB RAM, 12 core intel something. Failed with the following error. The query didn't complete in 5 minutes.

### Logs

```shell
2023-06-19T14:21:00+0200 [ERROR] [api_server:llm-falcon-service:19] Exception in ASGI application
Traceback (most recent call last):
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 428, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/uvicorn/middleware/message_logger.py", line 86, in __call__
    raise exc from None
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/uvicorn/middleware/message_logger.py", line 82, in __call__
    await self.app(scope, inner_receive, inner_send)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/bentoml/_internal/server/http/traffic.py", line 26, in __call__
    await self.app(scope, receive, send)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/bentoml/_internal/server/http/instruments.py", line 176, in __call__
    await self.app(scope, receive, wrapped_send)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/opentelemetry/instrumentation/asgi/__init__.py", line 579, in __call__
    await self.app(scope, otel_receive, otel_send)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/bentoml/_internal/server/http/access.py", line 126, in __call__
    await self.app(scope, receive, wrapped_send)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/starlette/_exception_handler.py", line 57, in wrapped_app
    raise exc
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/starlette/_exception_handler.py", line 46, in wrapped_app
    await app(scope, receive, sender)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/starlette/routing.py", line 727, in __call__
    await route.handle(scope, receive, send)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/starlette/routing.py", line 285, in handle
    await self.app(scope, receive, send)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/starlette/routing.py", line 74, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/starlette/_exception_handler.py", line 57, in wrapped_app
    raise exc
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/starlette/_exception_handler.py", line 46, in wrapped_app
    await app(scope, receive, sender)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/starlette/routing.py", line 69, in app
    response = await func(request)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/bentoml/_internal/server/http_app.py", line 286, in readyz
    runners_ready = all(await asyncio.gather(*runner_statuses))
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/bentoml/_internal/runner/runner.py", line 156, in runner_handle_is_ready
    return await self._runner_handle.is_ready(timeout)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/bentoml/_internal/runner/runner_handle/remote.py", line 304, in is_ready
    async with self._client.get(
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/aiohttp/client.py", line 1141, in __aenter__
    self._resp = await self._coro
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/aiohttp/client.py", line 560, in _request
    await resp.start(conn)
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/aiohttp/client_reqrep.py", line 894, in start
    with self._timer:
  File "/home/max/.conda/envs/openllm/lib/python3.10/site-packages/aiohttp/helpers.py", line 721, in __exit__
    raise asyncio.TimeoutError from None
asyncio.exceptions.TimeoutError
```

#### Environment variable

```bash
BENTOML_DEBUG=''
BENTOML_QUIET=''
BENTOML_BUNDLE_LOCAL_BUILD=''
BENTOML_DO_NOT_TRACK=''
BENTOML_CONFIG=''
BENTOML_CONFIG_OPTIONS=''
BENTOML_PORT=''
BENTOML_HOST=''
BENTOML_API_WORKERS=''
```

#### System information

`bentoml`: 1.0.22
`python`: 3.10.11
`platform`: Linux-5.4.0-42-generic-x86_64-with-glibc2.31
`uid_gid`: 1001:1001
`conda`: 4.14.0
`in_conda_env`: True
<details><summary><code>conda_packages</code></summary>

<br>

```yaml
name: openllm
channels:
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - _openmp_mutex=5.1=1_gnu
  - bzip2=1.0.8=h7b6447c_0
  - ca-certificates=2023.05.30=h06a4308_0
  - ld_impl_linux-64=2.38=h1181459_1
  - libffi=3.4.4=h6a678d5_0
  - libgcc-ng=11.2.0=h1234567_1
  - libgomp=11.2.0=h1234567_1
  - libstdcxx-ng=11.2.0=h1234567_1
  - libuuid=1.41.5=h5eee18b_0
  - ncurses=6.4=h6a678d5_0
  - openssl=3.0.8=h7f8727e_0
  - pip=23.1.2=py310h06a4308_0
  - python=3.10.11=h955ad1f_3
  - readline=8.2=h5eee18b_0
  - setuptools=67.8.0=py310h06a4308_0
  - sqlite=3.41.2=h5eee18b_0
  - tk=8.6.12=h1ccaba5_0
  - wheel=0.38.4=py310h06a4308_0
  - xz=5.4.2=h5eee18b_0
  - zlib=1.2.13=h5eee18b_0
  - pip:
    - accelerate==0.20.3
    - aiohttp==3.8.4
    - aiosignal==1.3.1
    - anyio==3.7.0
    - appdirs==1.4.4
    - asgiref==3.7.2
    - async-timeout==4.0.2
    - attrs==23.1.0
    - bentoml==1.0.22
    - build==0.10.0
    - cattrs==23.1.2
    - certifi==2023.5.7
    - charset-normalizer==3.1.0
    - circus==0.18.0
    - click==8.1.3
    - click-option-group==0.5.6
    - cloudpickle==2.2.1
    - cmake==3.26.4
    - coloredlogs==15.0.1
    - contextlib2==21.6.0
    - datasets==2.13.0
    - deepmerge==1.1.0
    - deprecated==1.2.14
    - dill==0.3.6
    - einops==0.6.1
    - exceptiongroup==1.1.1
    - filelock==3.12.2
    - filetype==1.2.0
    - frozenlist==1.3.3
    - fs==2.4.16
    - fsspec==2023.6.0
    - grpcio==1.54.2
    - grpcio-health-checking==1.48.2
    - h11==0.14.0
    - httpcore==0.17.2
    - httpx==0.24.1
    - huggingface-hub==0.15.1
    - humanfriendly==10.0
    - idna==3.4
    - importlib-metadata==6.0.1
    - inflection==0.5.1
    - jinja2==3.1.2
    - lit==16.0.6
    - markdown-it-py==3.0.0
    - markupsafe==2.1.3
    - mdurl==0.1.2
    - mpmath==1.3.0
    - multidict==6.0.4
    - multiprocess==0.70.14
    - mypy-extensions==1.0.0
    - networkx==3.1
    - numpy==1.25.0
    - nvidia-cublas-cu11==11.10.3.66
    - nvidia-cuda-cupti-cu11==11.7.101
    - nvidia-cuda-nvrtc-cu11==11.7.99
    - nvidia-cuda-runtime-cu11==11.7.99
    - nvidia-cudnn-cu11==8.5.0.96
    - nvidia-cufft-cu11==10.9.0.58
    - nvidia-curand-cu11==10.2.10.91
    - nvidia-cusolver-cu11==11.4.0.1
    - nvidia-cusparse-cu11==11.7.4.91
    - nvidia-nccl-cu11==2.14.3
    - nvidia-nvtx-cu11==11.7.91
    - openllm==0.1.6
    - opentelemetry-api==1.17.0
    - opentelemetry-instrumentation==0.38b0
    - opentelemetry-instrumentation-aiohttp-client==0.38b0
    - opentelemetry-instrumentation-asgi==0.38b0
    - opentelemetry-instrumentation-grpc==0.38b0
    - opentelemetry-sdk==1.17.0
    - opentelemetry-semantic-conventions==0.38b0
    - opentelemetry-util-http==0.38b0
    - optimum==1.8.8
    - orjson==3.9.1
    - packaging==23.1
    - pandas==2.0.2
    - pathspec==0.11.1
    - pillow==9.5.0
    - pip-requirements-parser==32.0.1
    - pip-tools==6.13.0
    - prometheus-client==0.17.0
    - protobuf==3.20.3
    - psutil==5.9.5
    - pyarrow==12.0.1
    - pydantic==1.10.9
    - pygments==2.15.1
    - pynvml==11.5.0
    - pyparsing==3.1.0
    - pyproject-hooks==1.0.0
    - pyre-extensions==0.0.29
    - python-dateutil==2.8.2
    - python-json-logger==2.0.7
    - python-multipart==0.0.6
    - pytz==2023.3
    - pyyaml==6.0
    - pyzmq==25.1.0
    - regex==2023.6.3
    - requests==2.31.0
    - rich==13.4.2
    - safetensors==0.3.1
    - schema==0.7.5
    - sentencepiece==0.1.99
    - simple-di==0.1.5
    - six==1.16.0
    - sniffio==1.3.0
    - starlette==0.28.0
    - sympy==1.12
    - tabulate==0.9.0
    - tokenizers==0.13.3
    - tomli==2.0.1
    - torch==2.0.1
    - torchvision==0.15.2
    - tornado==6.3.2
    - tqdm==4.65.0
    - transformers==4.30.2
    - triton==2.0.0
    - typing-extensions==4.6.3
    - typing-inspect==0.9.0
    - tzdata==2023.3
    - urllib3==2.0.3
    - uvicorn==0.22.0
    - watchfiles==0.19.0
    - wcwidth==0.2.6
    - wrapt==1.15.0
    - xformers==0.0.20
    - xxhash==3.2.0
    - yarl==1.9.2
    - zipp==3.15.0
prefix: /home/max/.conda/envs/openllm
```

</details>

<details><summary><code>pip_packages</code></summary>

<br>

```
accelerate==0.20.3
aiohttp==3.8.4
aiosignal==1.3.1
anyio==3.7.0
appdirs==1.4.4
asgiref==3.7.2
async-timeout==4.0.2
attrs==23.1.0
bentoml==1.0.22
build==0.10.0
cattrs==23.1.2
certifi==2023.5.7
charset-normalizer==3.1.0
circus==0.18.0
click==8.1.3
click-option-group==0.5.6
cloudpickle==2.2.1
cmake==3.26.4
coloredlogs==15.0.1
contextlib2==21.6.0
datasets==2.13.0
deepmerge==1.1.0
Deprecated==1.2.14
dill==0.3.6
einops==0.6.1
exceptiongroup==1.1.1
filelock==3.12.2
filetype==1.2.0
frozenlist==1.3.3
fs==2.4.16
fsspec==2023.6.0
grpcio==1.54.2
grpcio-health-checking==1.48.2
h11==0.14.0
httpcore==0.17.2
httpx==0.24.1
huggingface-hub==0.15.1
humanfriendly==10.0
idna==3.4
importlib-metadata==6.0.1
inflection==0.5.1
Jinja2==3.1.2
lit==16.0.6
markdown-it-py==3.0.0
MarkupSafe==2.1.3
mdurl==0.1.2
mpmath==1.3.0
multidict==6.0.4
multiprocess==0.70.14
mypy-extensions==1.0.0
networkx==3.1
numpy==1.25.0
nvidia-cublas-cu11==11.10.3.66
nvidia-cuda-cupti-cu11==11.7.101
nvidia-cuda-nvrtc-cu11==11.7.99
nvidia-cuda-runtime-cu11==11.7.99
nvidia-cudnn-cu11==8.5.0.96
nvidia-cufft-cu11==10.9.0.58
nvidia-curand-cu11==10.2.10.91
nvidia-cusolver-cu11==11.4.0.1
nvidia-cusparse-cu11==11.7.4.91
nvidia-nccl-cu11==2.14.3
nvidia-nvtx-cu11==11.7.91
openllm==0.1.6
opentelemetry-api==1.17.0
opentelemetry-instrumentation==0.38b0
opentelemetry-instrumentation-aiohttp-client==0.38b0
opentelemetry-instrumentation-asgi==0.38b0
opentelemetry-instrumentation-grpc==0.38b0
opentelemetry-sdk==1.17.0
opentelemetry-semantic-conventions==0.38b0
opentelemetry-util-http==0.38b0
optimum==1.8.8
orjson==3.9.1
packaging==23.1
pandas==2.0.2
pathspec==0.11.1
Pillow==9.5.0
pip-requirements-parser==32.0.1
pip-tools==6.13.0
prometheus-client==0.17.0
protobuf==3.20.3
psutil==5.9.5
pyarrow==12.0.1
pydantic==1.10.9
Pygments==2.15.1
pynvml==11.5.0
pyparsing==3.1.0
pyproject_hooks==1.0.0
pyre-extensions==0.0.29
python-dateutil==2.8.2
python-json-logger==2.0.7
python-multipart==0.0.6
pytz==2023.3
PyYAML==6.0
pyzmq==25.1.0
regex==2023.6.3
requests==2.31.0
rich==13.4.2
safetensors==0.3.1
schema==0.7.5
sentencepiece==0.1.99
simple-di==0.1.5
six==1.16.0
sniffio==1.3.0
starlette==0.28.0
sympy==1.12
tabulate==0.9.0
tokenizers==0.13.3
tomli==2.0.1
torch==2.0.1
torchvision==0.15.2
tornado==6.3.2
tqdm==4.65.0
transformers==4.30.2
triton==2.0.0
typing-inspect==0.9.0
typing_extensions==4.6.3
tzdata==2023.3
urllib3==2.0.3
uvicorn==0.22.0
watchfiles==0.19.0
wcwidth==0.2.6
wrapt==1.15.0
xformers==0.0.20
xxhash==3.2.0
yarl==1.9.2
zipp==3.15.0
```

</details>

- `transformers` version: 4.30.2
- Platform: Linux-5.4.0-42-generic-x86_64-with-glibc2.31
- Python version: 3.10.11
- Huggingface_hub version: 0.15.1
- Safetensors version: 0.3.1
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: ?
- Using distributed or parallel set-up in script?: ?

**Comment by aarnphm** [comment]
When did you see this timeout error?

**Comment by aarnphm** [comment]
I was able to run falcon with 4 T4s, but haven't able to run it with any lower number of GPU.

**Comment by maxfreu** [comment]
It appeared after the query was sent. I also tried with a 3090 with 24GB RAM. The GPU never sees pytorch launch, the nvidia-smi output is blank, CPU is at full power on all cores indefinitely.

**Comment by aarnphm** [comment]
Thats seems a weird bug, I will get on to this asap

**Comment by aarnphm** [comment]
I have identified the bug and will make a patch release soon cc @maxfreu

**Comment by aarnphm** [comment]
This has been fixed with 0.1.7, please install with the latest version. Thanks for finding the bug

**Comment by alew3** [comment]
I get the same error on a Mac M1 Max after I query the server. 
I start the server with: openllm start flan-t5 --model-id google/flan-t5-small
Tried pytorch and tensorflow and got the same error: 
[ERROR] [api_server:llm-flan-t5-service:6] Exception in ASGI application
